packages/api-derive/src/session/info.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright 2017-2019 @polkadot/api-derive authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { DerivedSessionInfo } from '../types';
+
+import BN from 'bn.js';
+import { Observable, combineLatest } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ApiInterface$Rx } from '@polkadot/api/types';
+import { BlockNumber, Option } from '@polkadot/types';
+
+import { drr } from '../util/drr';
+import { bestNumber } from '../chain';
+
+type Result = [BN, [BlockNumber?, Option<BlockNumber>?, BlockNumber?, BlockNumber?, BlockNumber?]];
```

Comment:
Would it be possible to make that the previous function always returns something (never undefined), so that we can remove all the `?`

afaict all the observables return something, so there should ideally never be `?` in return types.

Reply:
Nope, looking at it is is actually worse than that - the original (before api-derive) and before we reworked the API and before defaults returned `undefined` - so this is ancient code. :) Anyway, reworked so it is modern and working.